src/ui/customBreakpointsUI.ts

```diff
@@ -84,39 +120,52 @@ class BreakpointsDataProvider implements vscode.TreeDataProvider<Breakpoint | Ca
         return;
       }
 
+      session.customRequest('enableXHRBreakpoints', {
+        ids: this.xhrBreakpoints.filter(b => b.checkboxState).map(b => b.id),
+      });
+
       session.customRequest('setCustomBreakpoints', { ids: toEnable });
     });
   }
 
   /** @inheritdoc */
-  getTreeItem(item: Breakpoint | Category): vscode.TreeItem {
+  getTreeItem(item: Breakpoint | Category | XHRBreakpoint): vscode.TreeItem {
     return item;
   }
 
   /** @inheritdoc */
-  getChildren(item?: Breakpoint | Category): vscode.ProviderResult<(Breakpoint | Category)[]> {
+  getChildren(
+    item?: Breakpoint | Category | XHRBreakpoint,
+  ): vscode.ProviderResult<(Breakpoint | Category | XHRBreakpoint)[]> {
     if (!item) {
       return [...this.categories.values()];
     }
 
     if (item instanceof Category) {
+      if (item.contextValue === 'xhrCategory') {
+        return this.xhrBreakpoints.sort(XHRBreakpoint.compare);
```

Comment:
We could also have a tree item in the XHR category that would make a new breakpoint

```ts
const title = l10n.t('Add new URL');
const addNew = new vscode.TreeItem(title);
addNew.command = { title, command: Commands.AddXHRBreakpoints };
return [...this.xhrBreakpoints, addNew];
```